libs/langchain/langchain/document_transformers/dedup.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Iterable, Iterator, Sequence, Union
+
+from langchain.schema import BaseDocumentTransformer, Document
+from langchain.schema.document import _deduplicate_in_order, _HashedDocument
+
+
+class DedupDocumentTransformer(BaseDocumentTransformer):
+    def __init__(
+        self, by_content: bool = True, by_metadata: Union[bool, Sequence[str]] = False
+    ) -> None:
+        self.by_content = by_content
+        self.by_metadata = by_metadata
+
+    def _hashed_documents(
+        self, documents: Iterable[Document]
+    ) -> Iterator[_HashedDocument]:
+        for doc in documents:
+            page_content = doc.page_content if self.by_content else ""
+            if isinstance(self.by_metadata, Sequence):
```

Comment:
(fyi `str` is a sequence too -- not sure we should do anything here, but can become an issue if we get used to using it)